lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2212,15 +2189,16 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
     // inferred differentiation parameters.
     auto *original = cast<AbstractFunctionDecl>(match.Witness);
     auto *whereClauseGenEnv =
-        computeDerivativeGenericEnvironment(reqAttr, original);
+        reqAttr->computeDerivativeGenericEnvironment(original);
     auto *inferredParameters = TypeChecker::inferDifferentiableParameters(
         original, whereClauseGenEnv);
     bool omitWrtClause = reqAttr->getParameterIndices()->parameters.count() ==
                          inferredParameters->parameters.count();
     // Get `@differentiable` attribute description.
     std::string reqDiffAttrString;
     llvm::raw_string_ostream stream(reqDiffAttrString);
-    reqAttr->print(stream, req, omitWrtClause);
+    reqAttr->print(stream, req, omitWrtClause,
```

Comment:
Note: after https://github.com/apple/swift/pull/26113, protocol requirement `@differentiable` attributes will not be able to specify `jvp:` or `vjp:`.
So the `/*omitAssociatedFunctions*/ true` flag here doesn't change behavior for protocol requirements/witnesses.